File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        


        Set  <Integer> s = new HashSet<>();
        int l=0;
        int r;
        
        for(r=0;r < k+1; r++){
            if(r < nums.length ){if( s.contains(nums[r])) return true;
            s.add(nums[r]);}
        }
        for(r=k+1;r < nums.length;r++){
            s.remove(nums[l]);
            l++;
            if(s.contains(nums[r])) return true;
            s.add(nums[r]);
        }
        return false;

        
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        visited =set()
        l=0
        r=0
        for r in range(len(nums)):
            if r < k+1:
                if nums[r] in visited :
                    return True
                visited.add(nums[r])
            else :
                visited.remove(nums[l])
                l+=1
                if nums[r] in visited :
                    return True
                visited.add(nums[r])
        return False


        